platforms/extensibility/plugin-development/src/main/java/org/gradle/plugin/devel/internal/precompiled/GeneratePluginAdaptersTask.java

```diff
@@ -150,10 +149,17 @@ private PluginRequests extractPluginRequests(CompiledScript<PluginsAwareScript,
 
     private void generateScriptPluginAdapter(PrecompiledGroovyScript scriptPlugin, String firstPassCode) {
         String targetClass = scriptPlugin.getTargetClassName();
-        File outputFile = getPluginAdapterSourcesOutputDirectory().file(scriptPlugin.getPluginAdapterClassName() + ".java").get().getAsFile();
+        File outputFile = getPluginAdapterSourcesOutputDirectory().file(scriptPlugin.getPluginAdapterFullClassName().replace(".", "/") + ".java").get().getAsFile();
+
+        try {
+            Files.createDirectories(outputFile.getParentFile().toPath());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
 
         try (PrintWriter writer = new PrintWriter(Files.newBufferedWriter(Paths.get(outputFile.toURI())))) {
             writer.println("//CHECKSTYLE:OFF");
+            writer.println("package " + scriptPlugin.getPluginAdapterPackage() + ";");
```

Comment:
Is the generated class name considered stable? Are we allowed to change the package here?

I supposed we could also just suppress the warning with `@SuppressWarnings("DefaultPackage")`